lib/Dialect/SPIRV/SPIRVOps.cpp

```diff
@@ -199,6 +199,162 @@ static void printNoIOOp(Operation *op, OpAsmPrinter *printer) {
   printer->printOptionalAttrDict(op->getAttrs());
 }
 
+//===----------------------------------------------------------------------===//
+// spv.AccessChainOp
+//===----------------------------------------------------------------------===//
+
+// Function to extract an actual value from ConstantOp.
+static std::pair<StringRef, bool> extractValueFromConstOp(Operation *op,
```

Comment:
I think we just need to return a bool to indicating whether the extraction is successful. This is a utility function so it's up to the callers to decide what suitable error message will be good for their cases. (From this utility function's perspective, it should not know about struct type.)

Reply:
Thanks, updated.